react/forms/step4.md

```diff
@@ -0,0 +1,58 @@
+PatternFly components, including forms, can be customized with props. You can add a form validation effect to your cards with the `isValid` prop and helper text with the `helperTextInvalid` prop.
+
+## Task
+
+1) First, we'll update our constructor state declaration to have an additional `isValid` state prop:
+
+<pre class="file" data-target="clipboard">
+constructor(props) {
+  super(props);
+  this.state = {
+    value1: '',
+    value2: '',
+    value3: '',
+    value4: '',
+    isValid: false
+  };
+</pre>
+
+2) Next, we'll update `handleTextInputChange2` so it checks the email format against a regex:
+<pre class="file" data-target="clipboard">
+this.handleTextInputChange2 = value3 => {
+  const reg = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+  this.setState({ value3, isValid: reg.test(value3) });
+};
+</pre>
+
+3) Next, we'll add `isValid` to the state declaration in the `render()` method:
+
+<pre class="file" data-target="clipboard">
+render() {

```

Comment:
Again, I don't believe we should copy `render() {`, but show / help the user where to find it?